### PR TITLE
institutions: add institution literature table and papers count

### PR DESCRIPTION
**Pull request by monaawi**
* INSPIR-3312
* Needs institution remapping
* Needs alembic upgrade
* Needs remigration

### CODE REVIEW

File: backend/inspirehep/records/api/mixins.py [hunk 1]
```diff
@@ -233,3 +234,73 @@ def get_newest_linked_conferences_uuid(self):
...
+class InstitutionPapersMixin:
+    def clean_institution_literature_relations(self):
+        InstitutionLiterature.query.filter_by(literature_uuid=self.id).delete()
+
+    def create_institution_relations(self):
+        institutions = self.get_linked_records_from_field("authors.affiliations.record")
+        institution_literature_relations_waiting_for_commit = []
+
+        for institution in institutions:
+            if institution.get("deleted") is not True:
```
Comment (drjova) [inline backend/inspirehep/records/api/mixins.py:248]:
```suggestion
            if institution.get("deleted", False) is not True:
```

Comment (drjova) [reply backend/inspirehep/records/api/mixins.py:248]: We should have ``deleted`` property in base IMO, we're using it everywhere (not for this PR though)

File: backend/inspirehep/records/api/institutions.py [hunk 2]
```diff
@@ -16,3 +17,18 @@ class InstitutionsRecord(InspireRecord):
...
+
+    def delete_relations_with_literature(self):
+        InstitutionLiterature.query.filter_by(institution_uuid=self.id).delete()
+
+    def delete(self):
+        super().delete()
+        self.delete_relations_with_literature()
+
+    def hard_delete(self):
+        self.delete_relations_with_literature()
```
Comment (drjova) [inline backend/inspirehep/records/api/institutions.py:29]: AFAIS in the mixin you have already override the ``delete`` and ``hard_delete`` to delete the relations

File: backend/inspirehep/utils.py [hunk 3]
```diff
@@ -29,3 +29,12 @@ def chunker(iterable, chunksize):
...
         yield buf
+
+
+def flatten_list(input_list):
+    if isinstance(input_list, list):
+        return [
+            element for innerList in input_list for element in flatten_list(innerList)
+        ]
+    else:
+        return [input_list]
```
Comment (drjova) [inline backend/inspirehep/utils.py:40]:
```suggestion
    return [input_list]
```

File: backend/inspirehep/records/api/mixins.py (hunk 1 above)
Comment (monaawi) [reply backend/inspirehep/records/api/mixins.py:248]: but in this case, I don't necessarily need the default False

File: backend/inspirehep/records/api/institutions.py (hunk 2 above)
Comment (monaawi) [reply backend/inspirehep/records/api/institutions.py:29]: The mixin is on ```LiteratureRecord```. I need to delete the relations also on ```InstitutionRecord``` delete.

File: backend/tests/integration-async/migrator/test_migrator_tasks.py [hunk 4]
```diff
@@ -220,6 +261,39 @@ def test_update_relations(app, celery_app_with_context, celery_session_worker):
...
+            "affiliations": [{"value": "Institution", "record": {"$ref": inst_ref}}],
+        }
+    ]
+
+    record = InspireRecord.create(data, disable_relations_update=True)
+    db.session.commit()
+
+    task = update_relations.delay([record.id])
+
+    task.get(timeout=50)
```
Comment (drjova) [inline backend/tests/integration-async/migrator/test_migrator_tasks.py:288]: 50?

File: backend/tests/integration/records/api/test_api_institutions.py [hunk 5]
```diff
@@ -171,3 +172,81 @@ def test_aut_citation_count_property_blows_up_on_wrong_pid_type(base_app, db, es
 
     with pytest.raises(AttributeError):
         record.citation_count
+
+
+def test_deleted_institution_deletes_relations_in_institution_literature_table(
```
Comment (drjova) [inline backend/tests/integration/records/api/test_api_institutions.py:177]: Hmm I think we are missing the opposite; delete literature deletes relation with institution.

File: backend/tests/integration-async/migrator/test_migrator_tasks.py (hunk 4 above)
Comment (monaawi) [reply backend/tests/integration-async/migrator/test_migrator_tasks.py:288]: leftover

File: backend/tests/integration/records/api/test_api_institutions.py (hunk 5 above)
Comment (monaawi) [reply backend/tests/integration/records/api/test_api_institutions.py:177]: This is tested in the ```test_api_mixins``` https://github.com/inspirehep/inspirehep/pull/864/files#diff-b2fd3fb1e5f73a9192019311f091ae8aR168

**Pull request merged**